vms/evm/predicate/results_test.go

```diff
@@ -0,0 +1,128 @@
+		},
+		"multiple txs single result": {
+			results: map[common.Hash]TxResults{
+				{1}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+				},
+				{2}: map[common.Address][]byte{
+					{3}: {3, 2, 1},
+				},
+			},
+			expectedHex: "000000000002010000000000000000000000000000000000000000000000000000000000000000000001020000000000000000000000000000000000000000000003010203020000000000000000000000000000000000000000000000000000000000000000000001030000000000000000000000000000000000000000000003030201",
+		},
+		"multiple txs multiple results": {
+			results: map[common.Hash]TxResults{
+				{1}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+					{3}: {3, 2, 1},
+				},
+				{2}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+					{3}: {3, 2, 1},
+				},
+			},
+			expectedHex: "000000000002010000000000000000000000000000000000000000000000000000000000000000000002020000000000000000000000000000000000000000000003010203030000000000000000000000000000000000000000000003030201020000000000000000000000000000000000000000000000000000000000000000000002020000000000000000000000000000000000000000000003010203030000000000000000000000000000000000000000000003030201",
+		},
+		"multiple txs mixed results": {
+			results: map[common.Hash]TxResults{
+				{1}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+				},
+				{2}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+					{3}: {3, 2, 1},
+				},
+				{3}: map[common.Address][]byte{},
+			},
+			expectedHex: "000000000003010000000000000000000000000000000000000000000000000000000000000000000001020000000000000000000000000000000000000000000003010203020000000000000000000000000000000000000000000000000000000000000000000002020000000000000000000000000000000000000000000003010203030000000000000000000000000000000000000000000003030201030000000000000000000000000000000000000000000000000000000000000000000000",
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			require := require.New(t)
+			predicateResults := NewResultsFromMap(test.results)
+			b, err := predicateResults.Bytes()
+			require.NoError(err)
+			require.Equal(test.expectedHex, common.Bytes2Hex(b))
+
+			parsedPredicateResults, err := ParseResults(b)
+			require.NoError(err)
+			require.Equal(predicateResults, parsedPredicateResults)
+		})
+	}
+}
+
+func TestPredicateResultsAccessors(t *testing.T) {
+	require := require.New(t)
+
+	predicateResults := NewResults()
+
+	txHash := common.Hash{1}
+	addr := common.Address{2}
+	predicateResult := []byte{1, 2, 3}
+	txPredicateResults := map[common.Address][]byte{
+		addr: predicateResult,
+	}
+
+	require.Empty(predicateResults.GetResults(txHash, addr))
+	predicateResults.SetTxResults(txHash, txPredicateResults)
+	require.Equal(predicateResult, predicateResults.GetResults(txHash, addr))
+	predicateResults.DeleteTxResults(txHash)
+	require.Empty(predicateResults.GetResults(txHash, addr))
+
+	// Ensure setting empty tx predicate results removes the entry
```

Comment:
Having one mega test for all accessors isn't ideal - it's hard to extend the test with more cases and it's not clear which cases are being tested.. Can we have a separate test for `Get/Set` and `Get/Set/Delete`?